### ISSUE TITLE
update mutation not working on arrays

### ISSUE THREAD

**Issue by jamesfx2000** [Steams/ra-data-hasura-graphql#10]
I have a json field called `events` which contains a json array. 

When I try to update the array I get an error.

The mutation query sent to hasura contains `eventsIds` which causes the query to fail because that field doesn't exist. 

I tried looking through the source code but it seems mysterious to me. I'm not that familiar with graphql. 

I'd be happy to help if you could point me in the direction of some resources to learn.

**Comment by Steams** [comment]
There was some logic for nested object and array updates that i had not yet implemented. I'll set up a similar field and look into it

**Comment by Steams** [comment]
@jamesfx2000 Found the cause. When the response parser receives an array, it checks if the contents of the array are objects, and then it assumes that they must be references to other fields and will have an `id`, and so it returns a list of these id references,  `fieldnameIds`, to react admin. 
``` js
        if (Array.isArray(dataKey)) {
            if (typeof dataKey[0] === 'object') {
                return {
                    ...acc,
                    [key]: dataKey.map(sanitizeResource),
                    [`${key}Ids`]: dataKey.map(d => d.id),
                };
            } else {
                return { ...acc, [key]: dataKey };
            }
        }
```

When you then try to update a record, react admin sends the entire records data, which now includes this `fieldnameIds` property, which in your case doesn't exist. 

This `${key}Ids` property is something every other graphql data adapter I looked at uses. Instead of taking it out, i'll just add a check in the variablesBuilder so that it will only send that fieldIds property if it actually exists on the query.

**Comment by Steams** [comment]
Closed by 5373fb1b267a3262437a3fe6bb511e8bb9a4af47

**Comment by jamesfx2000** [comment]
@Steams 

Thanks!
